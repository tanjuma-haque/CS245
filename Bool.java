(0-1)*0.0 < 0.0
(0-1)*0.0 <= 0.0
4 < 5
9 >= 10
9 <= 10
"hi" < 8
"hi" < "hithere"
true > false
10 > 8
5 == 5
5 == 8.0
8 == 8.0
true == 1
false == true
(0-1)*0.0 == 0.0
"hello" == "hello"
"hi" != "hi"
null != false

3 < 4 && 5 < 10
3 < 4 && 5 == 10
3 < 4 || 5 == 10
7 >= 10 || 3 > 2
5 / 0 == 0 || 5 == 5
5 == 5 || 5 / 0 == 0
false || true && true
false || false && true
true || false && true
3 || true
