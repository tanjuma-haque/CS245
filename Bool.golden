Exit code: ExitSuccess
Stdout:
Welcome to the Java interpreter. Nice to meet you!
Type `:?` at the prompt for the list of commands.
Java> false
Java> true
Java> true
Java> false
Java> true
Java> String is not convertible to a numeric type.
Java> String is not convertible to a numeric type.
Java> boolean is not convertible to a numeric type.
Java> true
Java> true
Java> false
Java> true
Java> Illegal operand type(s) for equality: Boolean and Int
Java> false
Java> true
Java> false
Java> true
Java> Illegal operand type(s) for equality: <nulltype> and Boolean
Java> Java> true
Java> true
Java> true
Java> true
Java> ArithmeticException: / by zero
Java> true
Java> true
Java> false
Java> true
Java> Illegal operand type(s) for conditional operation: PrimType Int and PrimType Boolean
Java> Good-bye.

Stderr:
